tsconfig.json

```diff
@@ -18,7 +18,8 @@
       "typings",
       "node_modules/@types"
     ],
-    "esModuleInterop": true 
+    "esModuleInterop": true,
+    "noImplicitOverride": true,
```

Comment:
> Add noImplicitOverride to true in tsconfig, which may affect downstream packages with this setting enabled

any scenario where us enabling it here could cause type breaking changes to consumers? I couldn't think of one.

> My downstream has stricter typechecking, due to inclusion of [ts-reset](https://www.totaltypescript.com/ts-reset). While this package is not recommended to be used in packages, it would be nice for this package to be compatible with apps using this package.

I'm fine turning `noImplicitOverride` for stricter type, not sure for bigger changes just to be compatible with `ts-reset`.